### DISCUSSION TITLE
Is global variable g_ck available to custom slash commands?

### DISCUSSION THREAD

**Discussion by codaroma** [arnoudkooi/SN-Utils#489] [Ideas]
Is global variable g_ck available to custom slash commands?

For example it is needed for the query string in this URL

https://instance.service-now.com/export_update_set.do?sysparm_sys_id=aef00cad1bfcc250999062ca234bcb1d&sysparm_ck=f2c238bd1b5d4e5ca1fcecadee4bcbc32b49d15d85681d5d55fa4e9f86242c81220234b4&sysparm_is_remote=true

**Comment by arnoudkooi** [comment]
Not currently...
You could try with javascript: and window.open in the url field maybe?

**Reply by codaroma** [reply]
this works, thanks
```
javascript:void(open(
    "/export_update_set.do?sysparm_sys_id=$sysid&sysparm_is_remote=true&sysparm_ck=" +
        g_ck
));
```
The use of `void` is explained here https://stackoverflow.com/a/5412636/2530065

**Reply by arnoudkooi** [reply]
Cool, glad you figured it out and shared!
What is the use-case, export even when US not closed?

**Reply by codaroma** [reply]
It is for the sys_remote_update_set table. We have a lot of old previewed but not committed remote update sets on one of our instances that I would like to backup to XML and then remove. There is an OOTB form UI action called "Export to XML" but it only shows when the record is in the "Loaded" state and most instances auto-preview remote update sets when they are retrieved which makes the UI action hidden and useless. I was after a work-around that did not involve customising this OOTB UI action, or creating my own replacement UI action. I have enhanced it a bit more to add the update set name to the clipboard so I can paste it into the file name when saving. I have my browser set to always ask where to save download files, so I get a File Save As popup that I can then easily paste in the name of the update set from the clipboard. For some reason I had to encode the &<>" characters in the regex and URL otherwise they were converted to &amp;amp; &amp;lt; &amp;gt; &amp;quot; which breaks the regex and URL. This encoding of the & chars in the URL was not necessary on the previous version and I am not sure yet why it is needed with this version.
```
javascript:(function () {
    (window.gsft_main ? gsft_main.copyToClipboard : copyToClipboard)(
        g_form
            .getValue("name")
            .trim()
            .replace(/[\x3c\x3e:\x22/\\|?*]/g, "_") + " "
    );
    open(
        "/export_update_set.do?sysparm_sys_id=$sysid\x26sysparm_is_remote=true\x26sysparm_ck=" +
            g_ck
    );
})();
```

**Reply by codaroma** [reply]
BTW I think there is a bug in the way github encodes/processes comments. In order to type &amp;amp; I have to type it in as &amp;amp;amp; 
Which does not seem correct to me. 
Same with &amp;lt; 
I have to type this as &amp;amp;lt;
